internal/heartbeat/heartbeat.go

```diff
@@ -65,10 +73,7 @@ func (s *HeartbeatData) RetrieveInfo() models.Heartbeat {
 	config := s.configManager.GetDeviceConfiguration()
 	var hardwareInfo *models.HardwareInfo
 	if config.Heartbeat.HardwareProfile.Include {
-		hardwareInfo, err = s.hardware.GetHardwareInformation()
-		if err != nil {
-			log.Errorf("cannot get hardware information. DeviceID: %s; err: %v", s.workloadManager.GetDeviceID(), err)
-		}
+		hardwareInfo = s.buildHardwareInfo()
```

Comment:
@jakub-dzon why should the flow proceed if there was an error to collect HW info?

Reply:
IMO HW info are less important than receiving a heartbeat as if we do not send the heartbeat, there can be other implications at the other end of the tunnel (like disable workload or whatever) whereas the device is still UP
Nevertheless, if we let the flow proceed if a minor error occurs, we should notify it in the heartbeat